src/sql/src/plan/statement/show.rs

```diff
@@ -597,13 +599,18 @@ pub fn show_columns<'a>(
          WHERE mz_columns.id = '{}'",
         entry.id(),
     );
-    ShowSelect::new(
+    let (show_select, new_resolved_ids) = ShowSelect::new_with_resolved_ids(
```

Comment:
Why do we need `new_resolved_ids`?

Reply:
`new_resolved_ids` is the global ids that the transformed query is querying. If someone does `show columns t1` we turn that into `select ... from mz_catalog.mz_columns where id = 'u999'` which no longer references `t1` which is good b/c mz_support doesn't have read access on `t1`. Otoh, you can also do funky stuff like `show columns t1 where char_length(name) = any (select * from t1)` which gets desugared to `select ... from mz_catalog.mz_columns where id = 'u999' and char_length(name) = any (select * from t1)` which does read from `t1`.

Reply:
Ah ha! Good catch here, and thanks for explaining!